codegen-server/src/main/kotlin/software/amazon/smithy/rust/codegen/server/smithy/ValidateUnsupportedConstraints.kt

```diff
@@ -241,12 +243,12 @@ fun validateUnsupportedConstraints(
         .map { UnsupportedLengthTraitOnCollectionOrOnBlobShape(it, it.expectTrait()) }
         .toSet()
 
-    // 5. Range trait used on a non-integer shape. It has not been implemented yet.
+    // 5. Range trait used on unsupported shapes.
     // TODO(https://github.com/awslabs/smithy-rs/issues/1401)
     val unsupportedRangeTraitOnShapeSet = walker
         .walkShapes(service)
         .asSequence()
-        .filterNot { it is IntegerShape || it is ShortShape }
+        .filterNot { it is IntegerShape || it is ShortShape || it is LongShape || it is ByteShape }
         .filterMapShapesToTraits(setOf(RangeTrait::class.java))
         .map { (shape, rangeTrait) -> UnsupportedRangeTraitOnShape(shape, rangeTrait as RangeTrait) }
         .toSet()
```

Comment:
Do we still need this? I think `@range` on other shapes is rejected by smithy.

Reply:
`@range` is allowed on floats and doubles, according to the spec.

Reply:
You're right (and probably `bigDecimal` and `bigInteger`).
I'd suggest using `filter  { it is FloatShape || it is DoubleShape || .. }` instead of the negative as it makes it easier to see which are missing (but this is just a nit).